File: pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
 {
    
    if (argc != 2) {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    
    
    int key = atoi(argv[1]);

    if (key < 0) {
        printf("Usage: ./caesar key\n");
        return 1;
    } else {
        // prompt user for a code to encrypt
        string code = get_string("plaintext: ");
        printf("ciphertext: ");
        for (int i = 0, n = strlen(code); i < n; i++) {
          if islower(code[i]) {
            printf("%c", (((code[i] + key) - 97) % 26) + 97);
          } else if isupper(code[i]) {
            printf("%c", (((code[i] + key) - 65) % 26) + 65);
          } else {
            printf("%c", code[i]);
          }
        }
        printf("\n");
        return 0;
    }
 }

File: pset6/caesar/caesar.py
from sys import argv, exit
from cs50 import get_string


def main():
    # if args diff 2
    if len(argv) != 2:
        print("Usage: python caesar.py k")
    
    # key get second arg
    k = int(argv[1])
    # get string to cipher
    plaintext = get_string("plaintext: ")
    print("ciphertext: ", end="")
    
    # calculate ciphertext
    for ch in plaintext:
        if not ch.isalpha():
            print(ch, end="")
            continue

        ascii_offset = 65 if ch.isupper() else 97

        pi = ord(ch) - ascii_offset
        ci = (pi + k) % 26

        print(chr(ci + ascii_offset), end="")
        
    # prints newline
    print()
    
    # exit
    return 0


if __name__ == "__main__":
    main()